packages/grpc-native-core/index.d.ts

```diff
@@ -0,0 +1,1214 @@
+     *     function for outputs
+     * @param metadata Array of metadata key/value
+     *     pairs to add to the call
+     * @param options Options map
+     * @return An event emitter for stream related events
+     */
+    makeBidiStreamRequest(
+      method: string,
+      serialize: serialize,
+      deserialize: deserialize,
+      metadata?: Metadata | null,
+      options?: CallOptions | null,
+    ): ClientDuplexStream;
+
+    /**
+     * Close this client.
+     */
+    close(): void;
+
+    /**
+     * Return the underlying channel object for the specified client
+     * @return The channel
+     */
+    getChannel(): Channel;
+
+    /**
+     * Wait for the client to be ready. The callback will be called when the
+     * client has successfully connected to the server, and it will be called
+     * with an error if the attempt to connect to the server has unrecoverablly
+     * failed or if the deadline expires. This function will make the channel
+     * start connecting if it has not already done so.
+     * @param deadline When to stop waiting for a connection.
+     * @param callback The callback to call when done attempting to connect.
+     */
+    waitForReady(deadline: Deadline, callback: (error: Error | null) => void): void;
+  }
+
+  /**
+   * A gRPC channel.
+   */
+  export type Channel = any;
+
+  /**
+   * Options that can be set on a call.
+   */
+  export interface CallOptions {
+    /**
+     * The deadline for the entire call to complete.
+     */
+    deadline?: Deadline;
+    /**
+     * Server hostname to set on the call. Only meaningful if different from
+     * the server address used to construct the client.
+     */
+    host?: string;
+    /**
+     * Parent call. Used in servers when making a call as part of the process
+     * of handling a call. Used to propagate some information automatically,
+     * as specified by propagate_flags.
+     */
+    parent?: Call;
+    /**
+     * Indicates which properties of a parent call should propagate to this
+     * call. Bitwise combination of flags in `grpc.propagate`.
+     */
+    propagate_flags: number;
+    /**
+     * The credentials that should be used to make this particular call.
+     */
+    credentials: CallCredentials;
+  }
+
+  /**
+   * The deadline of an operation. If it is a date, the deadline is reached at
+   * the date and time specified. If it is a finite number, it is treated as
+   * a number of milliseconds since the Unix Epoch. If it is Infinity, the
+   * deadline will never be reached. If it is -Infinity, the deadline has already
+   * passed.
+   */
+  export type Deadline = number | Date;
+
+  /**
+   * Any client call type
+   */
+  type Call = ClientUnaryCall | ClientReadableStream | ClientWritableStream | ClientDuplexStream;
+
+  export class ClientUnaryCall {
+    /**
+     * An EventEmitter. Used for unary calls.
+     * @param call The call object associated with the request
+     */
+    constructor(call: any);
```

Comment:
As with the server-side call classes, I think these constructors should be omitted from the type definition.